### DISCUSSION TITLE
Is there any way to mix multiple versions in one project?

### DISCUSSION THREAD

**Discussion by vaynevayne** [fabricjs/fabric.js#10211] [Q&A]
Currently, the old code uses 4.5, and we want to upgrade to 6. How can I coexist two versions in one project?

**Comment by asturur** [comment]
Probably is not going to work. I would say no absolutely no

**Reply by vaynevayne** [reply]
Is this method okay? @asturur 
Known conditions: fabric installed via npm

[npmalias ](https://pnpm.io/aliases)
https://juejin.cn/post/7298241165526335497

**Reply by asturur** [reply]
If you are very careful that the code and imports do not get mixed maybe. But needs to be 2 different apps running in 2 contexts. Not the same app using fabric 4 and 6.

**Reply by vaynevayne** [reply]
> But needs to be 2 different apps running in 2 contexts.

I don't quite understand what app and context refer to in this sentence. Mine is a pc h5 page. In a window environment, I don't need to consider server-side rendering and other situations.
